src/Symfony/Component/Scheduler/Trigger/CronExpressionTrigger.php

```diff
@@ -25,20 +25,26 @@ final class CronExpressionTrigger implements TriggerInterface
 {
     public function __construct(
         private readonly CronExpression $expression = new CronExpression('* * * * *'),
+        private readonly int $randomDelay = 0
     ) {
     }
 
-    public static function fromSpec(string $expression = '* * * * *'): self
+    public static function fromSpec(string $expression = '* * * * *', int $randomDelay = 0): self
     {
         if (!class_exists(CronExpression::class)) {
             throw new LogicException(sprintf('You cannot use "%s" as the "cron expression" package is not installed; try running "composer require dragonmantank/cron-expression".', __CLASS__));
         }
 
-        return new self(new CronExpression($expression));
+        return new self(new CronExpression($expression), $randomDelay);
     }
 
     public function getNextRunDate(\DateTimeImmutable $run): ?\DateTimeImmutable
     {
-        return \DateTimeImmutable::createFromMutable($this->expression->getNextRunDate($run));
+        $nextRun = $this->expression->getNextRunDate($run);
+        if ($this->randomDelay > 0) {
+            $nextRun->add(new \DateInterval('PT'.random_int(0, $this->randomDelay).'S'));
```

Comment:
```suggestion
            $nextRun->add(new \DateInterval('PT'.random_int(-$this->randomDelay, $this->randomDelay).'S'));
```

Isn't it better to randomize the delay from -delay to +delay instead of 0?

Reply:
In this case, the value may be in the past and the task will be executed immediately. 
After receiving the task, `getNextRunDate` is called again and it may again in the past, therefore, the frequency of execution of the task may be violated